pytools/persistent_dict.py

```diff
@@ -479,7 +479,9 @@ class CollisionWarning(UserWarning):
 
 
 class _PersistentDictBase:
```

Comment:
Conceivably, we should make this a generic in terms of its key and value types.

Reply:
Should this derive from `MutableMapping`?

Reply:
Doesn't mypy figure out structurally whether something is a mapping?

Reply:
I'm not sure. I just realized that `Mapping` might not make too much sense here, since it the classes don't have `keys()` etc. It might make more sense in the context of #209.

Reply:
Is [f95b64c](https://github.com/inducer/pytools/pull/212/commits/f95b64c5a99f6fb9811cd38b30ad9016bc6426d2) what you had in mind? I couldn't find a way to make `WriteOnce...` generic as well, because then all consumers also would need to change their type annotations.

Reply:
Please just go ahead. I'm not worried about breaking downstream type annotations. They can always be fixed, and they don't prevent anyone from doing their work. At least mostly.

Reply:
What do you think of 2a8e506?